sds_data_manager/stacks/api_gateway_stack.py

```diff
@@ -0,0 +1,89 @@
+# Installed
+from constructs import Construct
+from aws_cdk import (
+    Stack,
+    Environment,
+    aws_apigateway as apigw,
+    aws_route53 as route53,
+    aws_certificatemanager as acm,
+    aws_route53_targets as targets
+)
+
+
+class ApiGateway(Stack):
+    """Sets up api gateway, creates subdomains, and creates methods that
+    are linked to the lambda function"""
+
+    def __init__(self, scope: Construct,
+                 construct_id: str,
+                 sds_id: str,
+                 lambda_functions: dict,
+                 env: Environment,
+                 hosted_zone: route53.IHostedZone = None,
+                 certificate: acm.ICertificate = None,
+                 use_custom_domain: bool = False,
+                 **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        sds_id : str
+            Name suffix for stack
+        lambda_functions : dict
+            Lambda functions
+        env : Environment
+        hosted_zone : route53.IHostedZone
+            Hosted zone used for DNS routing.
+        certificate : acm.ICertificate
+            Used for validating the secure connections to API Gateway.
+        use_custom_domain : bool
+            Determines if a custom domain should be used.
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # Define subdomains
+        subdomains = lambda_functions.keys()
+
+        for subdomain in subdomains:
```

Comment:
A short description in the code would be helpful. For example, how would API endpoint will look like after these operations. Would it look like (copying @greglucas above comment)

```
lambda1.dev.imap.com
lambda2.dev.imap.com
```
 or
```
api.dev.imap.com/lambda1
api.dev.imap.com/lambda2
```
or
```
imap-mission.com/dev/lambda1
imap-mission.com/dev/lambda2
```